src/platform/w3dengine/client/w3dview.cpp

```diff
@@ -790,3 +797,962 @@ void W3DView::Move_Along_Waypoint_Path(int milliseconds)
         }
     }
 }
+
+const Coord3D &W3DView::Get_3D_Camera_Position()
+{
+    static Coord3D pos;
+    Vector3 v = m_3DCamera->Get_Position();
+    pos.Set(v.X, v.Y, v.Z);
+    return pos;
+}
+
+void W3DView::Update_View()
+{
+#ifdef GAME_DEBUG_STRUCTS
+    // TODO Subsystem debug stuff
+#endif
+    Update();
+}
+
+float Get_Height_Around_Pos(float x, float y)
+{
+    float z1 = g_theTerrainLogic->Get_Ground_Height(x, y, nullptr);
+    float z2 = GameMath::Max(z1, g_theTerrainLogic->Get_Ground_Height(x + 40.0, y - 40.0, nullptr));
+    float z3 = GameMath::Max(z2, g_theTerrainLogic->Get_Ground_Height(x - 40.0, y - 40.0, nullptr));
+    float z4 = GameMath::Max(z3, g_theTerrainLogic->Get_Ground_Height(x + 40.0, y + 40.0, nullptr));
+    return GameMath::Max(z4, g_theTerrainLogic->Get_Ground_Height(x - 40.0, y + 40.0, nullptr));
+}
+
+void Draw_Drawable(Drawable *draw, void *user_data)
+{
+    draw->Draw(static_cast<View *>(user_data));
+}
+
+void W3DView::Update()
+{
+    static float follow_factor;
```

Comment:
This needs to have a default, otherwise first use may be undefined behavior.

Reply:
fixed.